spec/jobs/content_depositor_change_event_job_spec.rb

```diff
@@ -14,15 +13,38 @@
     allow(Time).to receive(:now).at_least(:once).and_return(mock_time)
   end
 
-  it "logs the event to the proxy depositor's profile, the depositor's dashboard, and the FileSet" do
-    expect do
-      described_class.perform_now(generic_work, another_user)
-    end.to change { user.profile_events.length }.by(1)
-                                                .and change { another_user.events.length }.by(1)
-                                                                                          .and change { generic_work.events.length }.by(1)
+  context "when use_valkyrie is false" do
+    let(:generic_work) { create(:generic_work, title: ['BethsMac'], user: user) }
 
-    expect(user.profile_events.first).to eq(event)
-    expect(another_user.events.first).to eq(event)
-    expect(generic_work.events.first).to eq(event)
+    it "logs the event to the proxy depositor's profile, the depositor's dashboard, and the FileSet" do
+      expect do
+        described_class.perform_now(generic_work, another_user)
+      end.to change { user.profile_events.length }.by(1)
+                                                  .and change { another_user.events.length }.by(1)
+                                                                                            .and change { generic_work.events.length }.by(1)
+
+      expect(user.profile_events.first).to eq(event)
+      expect(another_user.events.first).to eq(event)
+      expect(generic_work.events.first).to eq(event)
+    end
+  end
+
+  context "when use_valkyrie is true" do
+    let(:generic_work) { valkyrie_create(:hyrax_work, title: ['BethsMac']) }
+
+    it "logs the event to the proxy depositor's profile, the depositor's dashboard, and the FileSet" do
+      generic_work.apply_depositor_metadata(user)
+      allow_any_instance_of(described_class).to receive(:hyrax_test_simple_work_path).and_return("/concern/generic_works/#{generic_work.id}")
```

Comment:
do we need `allow_any_instance_of`? if we're stubbing this (+0 to that---if it gets around the other issues here, i'm okay with it), could we stub it on the current instance and call `#perform` directly instead of routing through `.perform_now`?